integration/airflow/openlineage/airflow/listener.py

```diff
@@ -0,0 +1,141 @@
+import logging
+import threading
+import uuid
+import attr
+
+from typing import TYPE_CHECKING, Optional, Callable
+
+from airflow.listeners import hookimpl
+
+from openlineage.airflow.adapter import OpenLineageAdapter
+from openlineage.airflow.extractors import ExtractorManager
+from openlineage.airflow.utils import DagUtils, get_task_location, get_job_name, get_custom_facets
+
+if TYPE_CHECKING:
+    from airflow.models import TaskInstance, BaseOperator
+    from sqlalchemy.orm import Session
+
+
+@attr.s(frozen=True)
+class RunData:
+    run_id: str = attr.ib()
+    task: "BaseOperator" = attr.ib()
+
+
+class RunDataHolder:
```

Comment:
To be consistent with `ExtractorManager` I would name this class `ActiveRunManager` or `ActiveTaskRunManager` with the interface:

```
add_active_run()
get_active_run()
```